File: java/program/Gerbong.java
public class Gerbong extends SaranaKereta {
    // Attributes
    private Trainmark jenisTrainmark;

    // Default Constructor
    public Gerbong() {
        super();
        this.jenisTrainmark = new Trainmark();
    }

    // Parameterized Constructor
    public Gerbong(String kodeSarana, String namaSarana, int beratSarana, String tipeBogie, Trainmark jenisTrainmark) {
        super(kodeSarana, namaSarana, beratSarana, tipeBogie);
        this.jenisTrainmark = jenisTrainmark;
    }

    // Getter and Setter for jenisTrainmark
    public Trainmark getJenisTrainmark() {return this.jenisTrainmark;}
    public void setJenisTrainmark(Trainmark trainmark) {this.jenisTrainmark = trainmark;}
}

File: cpp/program/Gerbong.cpp
#pragma once
#include "SaranaKereta.cpp"
#include "trainmark.cpp"

// Class Gerbong
class Gerbong : public SaranaKereta
{
// Attributes
private:
    Trainmark jenisTrainmark;

public:
    // Default Constructor
    Gerbong() : SaranaKereta() {
        jenisTrainmark = Trainmark();
    }

    // Parameterized Constructor
    Gerbong(string kodeSarana, string namaSarana, int beratSarana, string tipeBogie, Trainmark jenisTrainmark) : SaranaKereta(kodeSarana, namaSarana, beratSarana, tipeBogie) {
        this->jenisTrainmark = jenisTrainmark;
    }

    // Getter and Setter for jenisTrainmark
    Trainmark getJenisTrainmark() const {return this->jenisTrainmark;}
    void setJenisTrainmark(Trainmark trainmark) {this->jenisTrainmark = trainmark;}

    // Destructor
    ~Gerbong(){}
};